src/com/kodilla/controls/BoardPresenter.java

```diff
@@ -34,39 +59,72 @@ public BoardPresenter(BoardSettings settings) {
 
     @Override
     public void userMouseClicked(String userMark, int rowClicked, int columnClicked) {
-
+        computerControl = new ComputerControls();
+        prepareGameBoardToShow(userMark, rowClicked, columnClicked);
+        if (!ifWin) {
+            makeComputerMove();
+        }
     }
 
-    public void mouseClicked(String figureType, int rowClicked, int columnClicked) {
-        computerControl = new ComputerControls();
+    private void prepareGameBoardToShow(String gameMark, int rowClicked, int columnClicked) {
         int index = findIndexOfClickedElement(rowClicked, columnClicked);
-        setNewElementOnBoard(index, rowClicked, columnClicked, figureType);
-        makeComputerMove();
+        List<FieldState> listAfterClicked = setNewElementOnBoard(index, rowClicked, columnClicked, gameMark);
+        view.fillGameBoard(listAfterClicked);
+        winChecker(listAfterClicked, rowClicked, columnClicked, gameMark);
+        if (ifWin) {
+            winMessage();
```

Comment:
Najlepiej, aby metpda mowiła co robi. `winMessage` w tym momencie oznaczą wygraj wiadomość. Lepsza nazwa to: `showWinMessage()`

Reply:
done!